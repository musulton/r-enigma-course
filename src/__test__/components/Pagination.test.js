import {render, screen} from "@testing-library/react";
import userEvent from "@testing-library/user-event";

import Pagination from "../../components/Pagination";

describe("Pagination component", () => {
    it("should render Pagination correctly", () => {
        const mockProps = {
            currentPage: 1,
            totalPage: 10,
            onChangeCurrentPage: jest.fn()
        }

        const {asFragment} = render(<Pagination {...mockProps} />)

        expect(asFragment()).toMatchSnapshot();
    })

    it("should call onChangeCurrentPage with params when prev button is click and current page is not first page", () => {
        const mockOnChangeCurrentPage = jest.fn()
        const mockProps = {
            currentPage: 2,
            totalPage: 10,
            onChangeCurrentPage: mockOnChangeCurrentPage,
        }

        render(<Pagination {...mockProps} />)
        userEvent.click(screen.getByText("Prev"))

        expect(mockOnChangeCurrentPage).toBeCalledWith(mockProps.currentPage - 1)
    })

    it("should not call onChangeCurrentPage when prev button clicked and current page is first page", () => {
        const mockOnChangeCurrentPage = jest.fn()
        const mockProps = {
            currentPage: 1,
            totalPage: 10,
            onChangeCurrentPage: mockOnChangeCurrentPage,
        }

        render(<Pagination {...mockProps} />)
        userEvent.click(screen.getByText("Prev"))

        expect(mockOnChangeCurrentPage).not.toBeCalled()
    })

    it("should call onChangeCurrentPage with params when next button is click and current page is not last page", () => {
        const mockOnChangeCurrentPage = jest.fn()
        const mockProps = {
            currentPage: 1,
            totalPage: 10,
            onChangeCurrentPage: mockOnChangeCurrentPage,
        }

        render(<Pagination {...mockProps} />)
        userEvent.click(screen.getByText("Next"))

        expect(mockOnChangeCurrentPage).toBeCalledWith(mockProps.currentPage + 1)
    })

    it("should not call onChangeCurrentPage with params when prev button is click and current page is last page", () => {
        const mockOnChangeCurrentPage = jest.fn()
        const mockProps = {
            currentPage: 2,
            totalPage: 2,
            onChangeCurrentPage: mockOnChangeCurrentPage,
        }

        render(<Pagination {...mockProps} />)
        userEvent.click(screen.getByText("Next"))

        expect(mockOnChangeCurrentPage).not.toBeCalled()
    })
})
