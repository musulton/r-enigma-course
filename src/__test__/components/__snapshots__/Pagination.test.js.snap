// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination component should render Pagination correctly 1`] = `
<DocumentFragment>
  <ul
    class="sc-bgqQcB hkzmuC pagination"
  >
    <li
      class="page-item disabled"
    >
      <span
        class="page-link"
      >
        Prev
      </span>
    </li>
    <li
      class="page-item active"
    >
      <span
        class="page-link"
      >
        1
        <span
          class="visually-hidden"
        >
          (current)
        </span>
      </span>
    </li>
    <li
      class="page-item"
    >
      <a
        class="page-link"
        href="#"
        role="button"
        tabindex="0"
      >
        2
      </a>
    </li>
    <li
      class="page-item"
    >
      <a
        class="page-link"
        href="#"
        role="button"
        tabindex="0"
      >
        3
      </a>
    </li>
    <li
      class="page-item"
    >
      <a
        class="page-link"
        href="#"
        role="button"
        tabindex="0"
      >
        4
      </a>
    </li>
    <li
      class="page-item"
    >
      <a
        class="page-link"
        href="#"
        role="button"
        tabindex="0"
      >
        5
      </a>
    </li>
    <li
      class="page-item"
    >
      <a
        class="page-link"
        href="#"
        role="button"
        tabindex="0"
      >
        6
      </a>
    </li>
    <li
      class="page-item"
    >
      <a
        class="page-link"
        href="#"
        role="button"
        tabindex="0"
      >
        7
      </a>
    </li>
    <li
      class="page-item"
    >
      <a
        class="page-link"
        href="#"
        role="button"
        tabindex="0"
      >
        8
      </a>
    </li>
    <li
      class="page-item"
    >
      <a
        class="page-link"
        href="#"
        role="button"
        tabindex="0"
      >
        9
      </a>
    </li>
    <li
      class="page-item"
    >
      <a
        class="page-link"
        href="#"
        role="button"
        tabindex="0"
      >
        10
      </a>
    </li>
    <li
      class="page-item"
    >
      <a
        class="page-link"
        href="#"
        role="button"
        tabindex="0"
      >
        Next
      </a>
    </li>
  </ul>
</DocumentFragment>
`;
